hypothesis-python/src/hypothesis/provisional.py

```diff
@@ -51,17 +62,36 @@ def url_encode(s):
 
 
 @st.defines_strategy_with_reusable_values
-def domains():
-    """A strategy for :rfc:`1035` fully qualified domain names."""
+def domains(
+        max_length=None,  # type: Optional[int]
+        max_element_length=None  # type: Optional[int]
```

Comment:
For both arguments, the type signature and validation is simpler if we use integers as the default value.